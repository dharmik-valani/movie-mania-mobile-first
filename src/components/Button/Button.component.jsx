import React from 'react'

const Button = ({text,buttonclassName,textclassName,onClick }) => {
  return (
        <button className={buttonclassName} onClick={onClick} data-testid={text}>
            <span className={textclassName}>{text}</span>
        </button>
  )
}

export default Button