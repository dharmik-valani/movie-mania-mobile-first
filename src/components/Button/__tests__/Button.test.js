import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "../Button.component";

describe("Button", () => {
  let originalFetch;

  beforeEach(() => {
    originalFetch = global.fetch;
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve([
            {
                text: "View Details",
                buttonclassName: "border-amber-300 rounded-xl border-4 text-black py-2 px-4 rounded transition-opacity duration-300",
                textclassName: "text-white"
            },
          ]),
      })
    );
  });

  afterEach(() => {
    global.fetch = originalFetch;
  });

  it("Renders Button component", async () => {
    
    let propData = {
        text: "View Details",
        buttonclassName: "border-amber-300 rounded-xl border-4 text-black py-2 px-4 rounded transition-opacity duration-300",
        textclassName: "text-white"
    } 

    
    render(<Button {...propData}/>);
    const linkElement = await screen.getByTestId('View Details');
    expect(linkElement).toBeInTheDocument();
  });

  //  SNAP SHOT
  test("matches button snapshot", async () => {
    let propData = {
        text: "View Details",
        buttonclassName: "border-amber-300 rounded-xl border-4 text-black py-2 px-4 rounded transition-opacity duration-300",
        textclassName: "text-white"
    } 
    const tree = await renderer.create(<Button {...propData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});


