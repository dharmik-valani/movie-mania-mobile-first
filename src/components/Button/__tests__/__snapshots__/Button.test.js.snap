// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches button snapshot 1`] = `
<button
  className="border-amber-300 rounded-xl border-4 text-black py-2 px-4 rounded transition-opacity duration-300"
  data-testid="View Details"
>
  <span
    className="text-white"
  >
    View Details
  </span>
</button>
`;
